rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_subscriber_info.hpp

```diff
@@ -142,7 +142,7 @@ class SubListener : public EventListenerInterface, public eprosima::fastrtps::Su
   }
 
   void
-  data_taken(eprosima::fastrtps::Subscriber * sub)
+  updateDataCount(eprosima::fastrtps::Subscriber * sub)
```

Comment:
it seems to be intentional to call this `data_taken` after if the data is taken from RTPSReader.

Reply:
I think so.

But from the [sample](https://github.com/iuhilnehc-ynos/ros2_issues_test.git), it seems there is a case as follow,
```
FastDDS -> rmw_fastrtps_cpp : there is a data changed coming, set a flag

rclcpp
   spin
      rcl_wait(rmw_wait) -> check the flag
      get_data()
          if true, data_taken() -> reset the flag

But If the data changed removed not in get_data(), rclcpp::spin will get the data again and again because no function will reset the flag
```
So I think `data_taken` might be renamed as `updateDataCount`, and if get_data() failed, it also call `updateDataCount`.